light-client/relayer/tests/block_processing.rs

```diff
@@ -0,0 +1,191 @@
+        .expect("Could not create watcher_db");
+
+    // Initialize ledger
+    let sender = AccountKey::random(&mut rng);
+    let num_blocks = 1;
+    initialize_ledger(BLOCK_VERSION, &mut ledger, num_blocks, &sender, &mut rng);
+
+    // Make node signer identities
+    let signers: Vec<Ed25519Pair> = (0..5).map(|_| Ed25519Pair::from_random(&mut rng)).collect();
+
+    let config = Config {
+        start_block_index: 1,
+        min_signatures: 4,
+        ledger_db: ledger_db_path,
+        watcher_db: watcher_db_path,
+        admin_listen_uri: None,
+    };
+
+    log::info!(logger, "Starting relayer");
+
+    let runner = Relayer::new(
+        config,
+        ledger.clone(),
+        watcher.clone(),
+        DummySender {
+            logger: logger.clone(),
+        },
+        logger.clone(),
+    );
+
+    // Do happy path testing where there is a burn txo
+    let burn_txo = TxOut::new_with_memo(
+        BLOCK_VERSION,
+        Amount::new(100, 2.into()),
+        &burn_address(),
+        &FromRandom::from_random(&mut rng),
+        EncryptedFogHint::fake_onetime_hint(&mut rng),
+        |_ctxt| Ok(BurnRedemptionMemo::new([7u8; 64]).into()),
+    )
+    .unwrap(); // TODO: Use real format here?
+
+    let random_txo = TxOut::new(
+        BLOCK_VERSION,
+        Amount::new(100, 2.into()),
+        &PublicAddress::from_random(&mut rng),
+        &FromRandom::from_random(&mut rng),
+        EncryptedFogHint::fake_onetime_hint(&mut rng),
+    )
+    .unwrap();
+
+    log::info!(logger, "Adding block");
+
+    let (block, block_contents, block_metadata) =
+        append_tx_outs_as_block(&mut ledger, &[burn_txo.clone(), random_txo], signers);
+
+    log::info!(logger, "Adding block to watcher");
+
+    // Add this stuff to the watcher
+    for (idx, block_metadata) in block_metadata.iter().enumerate() {
+        let block_data = BlockData::new(
+            block.clone(),
+            block_contents.clone(),
+            None,
+            Some(block_metadata.clone()),
+        );
+
+        let url = Url::from_str(&format!("https://node{}.localhost", idx)).unwrap();
+
+        watcher.add_block_data(&url, &block_data).unwrap();
+    }
+
+    let mut retries = 100;
+    loop {
+        if retries == 0 {
+            panic!("relayer message not received");
+        }
+        retries -= 1;
+        let records = runner.get_burned_tx_records();
+        if records.len() > 0 {
+            assert_eq!(records.len(), 1);
+            let burn_record = records[0].clone();
+            assert_eq!(burn_record.tx_outs, vec![burn_txo.clone()]);
+            assert_eq!(burn_record.block.id, block.id);
+            assert_eq!(burn_record.block.contents_hash, block.contents_hash);
+            assert_eq!(burn_record.block_contents, block_contents);
+
+            let burn_record_signatures_count = burn_record.signatures.iter().count();
+            let block_signatures_count = block_metadata.iter().count();
```

Comment:
:upside_down_face: nit, Since these are vectors I think they can use `len()`, doesn't matter too much as it's a test
```suggestion
            let burn_record_signatures_count = burn_record.signatures.len();
            let block_signatures_count = block_metadata.len();
```